src/components/TokenSelect/index.tsx

```diff
@@ -133,18 +133,8 @@ export default memo(function TokenSelect({ collapsed, disabled, field, onSelect,
 
   const [open, setOpen] = useState(false)
   const onTokenSelectorClick = useAtomValue(onTokenSelectorClickAtom)
-  const onOpen = useCallback(() => {
-    const promise = onTokenSelectorClick?.(field)
-    if (promise) {
-      return promise
-        .then((open) => {
-          setOpen(open)
-        })
-        .catch(() => {
-          setOpen(false)
-        })
-    }
-    return setOpen(true)
+  const onOpen = useCallback(async () => {
+    setOpen((await onTokenSelectorClick?.(field)) ?? true)
```

Comment:
I'm pretty sure this drops the `setOpen(false)` on promise rejection functionality

Reply:
you're right, I'll back these changes out

Reply:
this was caught by tests, too :)